### PR TITLE
[9.0] [port] base_transaction_id

### PR DESCRIPTION
**Pull request by yvaucher**

### CODE REVIEW

**Comment by yvaucher** [comment]
lint is complaining about `from opererp.osv import expression` which I disagree on depreciacion

**Comment by pedrobaeza** [comment]
@yvaucher, this is only a warning. Real problems are:

```
base_transaction_id/models/account_move.py:37:9: F841 local variable 'prepare' is assigned to but never used
base_transaction_id/models/account_move.py:40:21: F821 undefined name 'lines'
```

**Comment by yvaucher** [comment]
@pedrobaeza Thx, I fixed it.

File: base_transaction_id/README.rst [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+Credits
+=======
+
+Contributors
+------------
+
+* Yannick Vaucher <yannick.vaucher@camptocamp.com>
+* Joël Grand-Guillaume <joel.grandguillaume@camptocamp.com>
+* Alexandre Fayolles <alexandre.fayolles@camptocamp.com>
```
Comment (guewen) [inline base_transaction_id/README.rst]: s/Fayolles/Fayolle/

File: base_transaction_id/README.rst [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+* Guewen Baconnier <guewen.baconnier@camptocamp.com>
+* Pedra Baeza <pedro.baezo@gmail.com>
```
Comment (guewen) [inline base_transaction_id/README.rst]: s/Pedra/Pedro/

**Comment by yvaucher** [comment]
Thx @guewen Fixed those typo

**Comment by yvaucher** [comment]
Currently, the js inheritance needs a fixup

(it does a silent error and makes the view unloadable (blank screen with Odoo logo and top menu bar without menus))

**Comment by yvaucher** [comment]
And we are good

**Comment by yvaucher** [comment]
travis is not starting ?

File: .travis.yml [hunk 3]
```diff
@@ -8,9 +8,12 @@ cache: pip
 
 addons:
   apt:
+   sources:
+      - pov-wkhtmltopdf
```
Comment (guewen) [inline .travis.yml:12]: nitpicking: `sources` is not aligned with `packages`

**Comment by guewen** [comment]
No idea why travis does not start...

**Comment by yvaucher** [comment]
@guewen nitpicking fixed, and travis is starting this was perhaps blocking it?

**Comment by guewen** [comment]
:+1:

**Comment by yvaucher** [comment]
bump

**Comment by yvaucher** [comment]
I did some fixes and ensured last commits in 8.0 were there.

This PR is ready

File: base_transaction_id/models/sale.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
```
Comment (leemannd) [inline base_transaction_id/models/sale.py]: short header here

File: base_transaction_id/models/account_move.py [hunk 5]
```diff
@@ -17,40 +16,46 @@
 #    You should have received a copy of the GNU Affero General Public License
 #    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
```
Comment (leemannd) [inline base_transaction_id/models/account_move.py]: Short header

File: base_transaction_id/models/invoice.py [hunk 6]
```diff
@@ -17,8 +16,6 @@
 #    You should have received a copy of the GNU Affero General Public License
 #    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
```
Comment (leemannd) [inline base_transaction_id/models/invoice.py]: Short header

**Comment by leemannd** [comment]
Nice work. Just cosmetic modifications in my comments.

File: base_transaction_id/models/account_move.py [hunk 7]
```diff
@@ -17,40 +16,46 @@
...
-##############################################################################
-from openerp.osv import orm, fields
+from openerp import models, fields, api
+from openerp.osv import expression
```
Comment (leemannd) [inline base_transaction_id/models/account_move.py:5]: from openerp.models import expression ?

Comment (yvaucher) [inline base_transaction_id/models/account_move.py:5]: I prefer direct import than using an alias

**Comment by yvaucher** [comment]
Fixed license headers

File: base_transaction_id/models/account_move.py (hunk 7 above)
Comment (leemannd) [inline base_transaction_id/models/account_move.py:5]: Seems legit

**Comment by leemannd** [comment]
:+1:

**Pull request merged**